datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -344,7 +344,7 @@ func (f *ChainDataFetcher) updateGauge(insert func(chainEvent blockchain.ChainEv
 func (f *ChainDataFetcher) retryFunc(insert func(blockchain.ChainEvent) error, gauge metrics.Gauge) func(blockchain.ChainEvent) {
 	return func(event blockchain.ChainEvent) {
 		i := 0
-		for err := insert(event); err != nil; {
+		for err := insert(event); err != nil; err = insert(event) {
```

Comment:
This line would be the fix, do I understand correctly?

Reply:
Yes, it is a bug fix. It didn't actually retry the `insert` function ;;